tests/lib/rook.py

```diff
@@ -122,20 +122,23 @@ def build_rook(self):
         )
         execute(
             "git clone https://github.com/rook/rook.git %s"
-            % os.path.join(self.builddir, 'src/github.com/rook/rook')
+            % os.path.join(self.builddir, 'src/github.com/rook/rook'),
+            log_stderr=False
```

Comment:
Do we still see the error in case of a failure?

Reply:
Not at the moment.

Should we retroactively capture stdin/stdout on failure? (ie if check=True, and we get a failure, we always capture regardless if capture=True is set?)

Then we could log the output on failure. Otherwise at the moment it's up to the user of execute to try/except the CalledProcessError and decide what to display.

I might follow this one up later.

Reply:
Okay, so I looked into this and I'm not sure if we want to return it on failure. Primarily the problem is that we'd always have to capture the output which could be very expensive and possibly useless. Therefore I think the caller of execute should decide if they want to supply capture=True and handle the output in the raised exception as appropriate.